### PR TITLE
stdenv: start deprecating non-list mesonFlags

### PR DESCRIPTION
**Pull request by Artturin**
the motivation for this is to simplify stdenv and ease the job of
reviewers due to them needing to tell contributors about the defacto
rule that mesonFlags should be a list of strings

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-burns** [comment]
Looks like there is a string-typed mesonFlags in speech-denoiser causing the ofborg eval error

File: pkgs/stdenv/generic/make-derivation.nix [hunk 1]
```diff
@@ -361,26 +361,26 @@ else let
...
-    } // lib.optionalAttrs (__contentAddressed) {
-      inherit __contentAddressed;
-      # Provide default values for outputHashMode and outputHashAlgo because
-      # most people won't care about these anyways
-      outputHashAlgo = attrs.outputHashAlgo or "sha256";
-      outputHashMode = attrs.outputHashMode or "recursive";
-    } // lib.optionalAttrs (stdenv.hostPlatform != stdenv.buildPlatform) {
-      mesonFlags = if mesonFlags == null then null else let
-        # See https://mesonbuild.com/Reference-tables.html#cpu-families
         cpuFamily = platform: with platform;
```
Comment (jtojnar) [inline pkgs/stdenv/generic/make-derivation.nix:377]: Needs to be indented one deeper.

File: pkgs/stdenv/generic/make-derivation.nix [hunk 2]
```diff
@@ -361,26 +361,26 @@ else let
...
-      inherit outputs;
-    } // lib.optionalAttrs (__contentAddressed) {
-      inherit __contentAddressed;
-      # Provide default values for outputHashMode and outputHashAlgo because
-      # most people won't care about these anyways
-      outputHashAlgo = attrs.outputHashAlgo or "sha256";
-      outputHashMode = attrs.outputHashMode or "recursive";
-    } // lib.optionalAttrs (stdenv.hostPlatform != stdenv.buildPlatform) {
-      mesonFlags = if mesonFlags == null then null else let
-        # See https://mesonbuild.com/Reference-tables.html#cpu-families
```
Comment (Mindavi) [inline pkgs/stdenv/generic/make-derivation.nix:377]: Do keep this URL

**Review by magnetophon — APPROVED** [review]
speech-denoiser LGTM, thanks!

**Review by jtojnar — APPROVED** [review]

**Pull request merged**